starlette/testclient.py

```diff
@@ -113,14 +127,21 @@ async def _run(self) -> None:
         """
         The sub-thread in which the websocket session runs.
         """
-        scope = self.scope
-        receive = self._asgi_receive
-        send = self._asgi_send
-        try:
-            await self.app(scope, receive, send)
-        except BaseException as exc:
-            self._send_queue.put(exc)
-            raise
+
+        async def run_app(tg: anyio.abc.TaskGroup) -> None:
+            try:
+                await self.app(self.scope, self._asgi_receive, self._asgi_send)
+            except anyio.get_cancelled_exc_class():
+                ...
+            except BaseException as exc:
+                self._send_queue.put(exc)
+                raise
+            tg.cancel_scope.cancel()
```

Comment:
Shouldn't we cancel the task group before raising the exception just above?

Reply:
I did that now.

Reply:
Thanks.